intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/formatting/DefaultXtextBlock.xtend

```diff
@@ -0,0 +1,227 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.idea.formatting
+
+import com.google.common.collect.LinkedListMultimap
+import com.google.inject.Inject
+import com.google.inject.Injector
+import com.intellij.formatting.ASTBlock
+import com.intellij.formatting.Alignment
+import com.intellij.formatting.Block
+import com.intellij.formatting.ChildAttributes
+import com.intellij.formatting.Indent
+import com.intellij.formatting.SpacingBuilder
+import com.intellij.formatting.Wrap
+import com.intellij.lang.ASTNode
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiComment
+import com.intellij.psi.formatter.common.AbstractBlock
+import java.util.LinkedList
+import java.util.List
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.ide.editor.bracketmatching.BracePair
+import org.eclipse.xtext.ide.editor.bracketmatching.IBracePairProvider
+
+import static extension com.intellij.psi.formatter.FormatterUtil.*
+import static extension com.intellij.psi.impl.source.tree.TreeUtil.*
+
+/**
+ * @author kosyakov - Initial contribution and API
+ */
+class DefaultXtextBlock extends AbstractBlock implements ModifiableBlock {
+
+	static val char LINE_FEED = '\n'
```

Comment:
It seems that IDEA abstracts away from platform specific line separators and inserts them only on save, as a client we should always use only '\n'.

Look at 'What are the rules of working with it?' here: https://confluence.jetbrains.com/display/IDEADEV/IntelliJ+IDEA+Architectural+Overview